src/containers/projectcontainer.js

```diff
@@ -29,46 +35,67 @@ class ProjectContainer extends Component {
     }
   }
 
-
-  onProjectUpdate = (data) => {
-    if (this.props.project.data === null) {
+  onSubmit = (e) => {
+    e.preventDefault();
+    const data = this.props.form.data;
+    // TODO: Use selected_project instead of props.params.id?
+    const id = this.props.params.id;
+    if (id === undefined) {
       this.props.createProject(data)
         .then(p => this.context.router.push(`/projects/${p.payload.id}`));
     } else {
-      this.props.updateProject(this.props.project.data.id, data)
-        .then(() => this.context.router.push(`/projects/${this.props.project.data.id}`));
+      this.props.updateProject(id, data);
     }
   }
 
+  onChange = (fieldName, value) => {
+    this.props.updateField(fieldName, value);
+  }
+
   render() {
-    return (
-      <Project
-        project={this.props.project}
-        customers={this.props.customers}
-        onProjectUpdate={this.onProjectUpdate}
-      />
-    );
+    const isNew = this.props.params.id === undefined;
+
+    if (this.props.customers.loading === true ||
+        this.props.form.data === undefined ||
```

Comment:
Agreed. That can't happen anymore
20. jun. 2016 14:14 skrev "Erlend Hamberg" <notifications@github.com>:

> In src/containers/projectcontainer.js
> <https://github.com/blankoslo/floq-projects/pull/3#discussion_r67679383>:
>
> >    render() {
> > -    return (
> > -      <Project
> > -        project={this.props.project}
> > -        customers={this.props.customers}
> > -        onProjectUpdate={this.onProjectUpdate}
> > -      />
> > -    );
> > +    const isNew = this.props.params.id === undefined;
> > +
> > +    if (this.props.customers.loading === true ||
> > +        this.props.form.data === undefined ||
>
> Can this happen? The initial state of form (in the reducer) isn't
> undefined.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/blankoslo/floq-projects/pull/3/files/7b0b11d656ef5c2cdb96a00d3dc9cb50ee9b4b27#r67679383>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABCBjCUeUPrQPxMtFiWtq-QAMXXmhp4gks5qNoQ1gaJpZM4I5nYD>
> .
>